### PR TITLE
[sdarq] Initial app questionnaire

### PR DESCRIPTION
**Pull request by zbedo**
This PR introduces:

- Security checklist and questionnaire for the new apps that will be deployed

### CODE REVIEW

File: sdarq/frontend/src/app/app-form/form.json [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+                        {
+                            "value": "azure_gcp",
+                            "text": "Both"
+                        }
+                    ]
+                },
+                {
+                    "type": "radiogroup",
+                    "name": "Architecture",
+                    "title": "Your service must be based on one of the following",
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:88]:
```suggestion
                    "title": "On which service is your app based (must be one of the following)?",
```

File: sdarq/frontend/src/app/app-form/form.json [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+                            "value": "Secret_management",
+                            "text": "We are currently using tools to store our secrets"
+                        },
+                        {
+                            "value": "Monitoring_tool",
+                            "text": " We are currently using Sentry or a similar tool (ex: Google Debug) for exception handling and reporting"
+                        },
+                        {
+                            "value": "SAST",
+                            "text": "We are currently using Codacy, SonacCloud or similar tools for security static code analysis"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:127]:
```suggestion
                            "text": "We are currently using Codacy, SonarCloud or similar tools for security static code analysis"
```

File: sdarq/frontend/src/app/app-form/form.json [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+                        },
+                        {
+                            "value": "Image_scan",
+                            "text": "We are currently using tools to scan our Docker images"
+                        },
+                        {
+                            "value": "DAST",
+                            "text": "Our service is currently scanned weekly/monthly via ZAP or similar tools (if there is a static IP)"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:135]: Should there be an extra related question such as "Our service is deployed to an environment at a static URL"

File: sdarq/frontend/src/app/app-form/form.json [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+                        }
+                    ]
+                },
+                {
+                    "type": "checkbox",
+                    "name": "Applicable",
+                    "title": "Check what's applicable",
+                    "choices": [{
+                            "value": "managed-sql",
+                            "text": "I am using managed SQL database"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:145]:
```suggestion
                            "text": "I am using a managed SQL database"
```

File: sdarq/frontend/src/app/app-form/form.json [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+                        },
+                        {
+                            "value": "in-cluster-sql",
+                            "text": "I am using in-cluster SQL database"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:149]:
```suggestion
                            "text": "I am using an in-cluster SQL database"
```

File: sdarq/frontend/src/app/app-form/form.json [hunk 6]
```diff
@@ -0,0 +1,580 @@
...
+                    "type": "html",
+                    "name": "Alert-sentry",
+                    "visibleIf": "{Security tools} notcontains \"Monitoring_tool\"",
+                    "html": "<div class=\"highs\">You should setup\n Sentry or a similar tool. Reach out to #dsp-devops to get started.</div>"
+                },
+                {
+                    "type": "html",
+                    "name": "Alert-codacy",
+                    "visibleIf": "{Security tools} notcontains \"SAST\"",
+                    "html": "<div class=\"highs\">You should use\n  Codacy, SonacCloud or similar tools for security static code analysis.\n  Reach out to Appsec team to set it up for your repository.\n</div>"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:380]:
```suggestion
                    "html": "<div class=\"highs\">You should use\n  Codacy, SonarCloud or similar tools for security static code analysis.\n  Reach out to Appsec team to set it up for your repository.\n</div>"
```

File: sdarq/frontend/src/app/app-form/form.json [hunk 7]
```diff
@@ -0,0 +1,580 @@
...
+                    "type": "html",
+                    "name": "Alert-database",
+                    "visibleIf": "{Applicable} contains \"managed-sql\"",
+                    "html": "<div class=\"highs\">Please make sure your database is using a private IP or is behind a proxy.</div>"
+                },
+                {
+                    "type": "html",
+                    "name": "Alert-Docker-images-update",
+                    "visibleIf": "{docker_images_update} = \"No\"",
+                    "html": "<div class=\"highs\">Reach out to AppSec team to discuss about the DOcker images update plan.</div>"
```
Comment (TomConner) [inline sdarq/frontend/src/app/app-form/form.json:392]:
```suggestion
                    "html": "<div class=\"highs\">Reach out to AppSec team to discuss about the Docker image update plan.</div>"
```

**Review by TomConner — CHANGES_REQUESTED** [review]
Looks really good. Most of my comments were minor. I did not check against the app checklist created during a recent meeting (I think Sarah was the author).

File: sdarq/frontend/src/app/app-form/form.json (hunk 3 above)
Comment (zbedo) [reply sdarq/frontend/src/app/app-form/form.json:135]: I think, for now, let's keep it like this, we still don't have a lot of information about apps. Maybe we add it in the future.

**Comment by zbedo** [comment]
> Looks really good. My comments were minor. I did not check against the app checklist created during a recent meeting (I think Sarah was the author).

Yes, this is 3rd party Apps questionnaire in Sdarq as part of the App Production Checklist that we discussed with the dev teams. (As it is Sdarq-new service questionnaire part of the `Security&Compliance` section in the New Service production readiness checklist). This questionnaire does not replace App Production Checklist, it is part of it.

**Pull request merged**